lib/kafo/puppet_command.rb

```diff
@@ -44,11 +38,65 @@ def self.search_puppet_path(bin_name)
 
     private
 
+    def manifest
+      %{echo '
+        $kafo_config_file="#{@configuration.config_file}"
+        #{add_progress}
+        #{generate_version_checks.join("\n") if @puppet_version_check}
+        #{@command}
+      '}
+    end
+
+    def add_progress
+      %{$kafo_add_progress="#{!KafoConfigure.verbose}"}
+    end
+
+    def generate_version_checks
+      checks = []
+      modules_path.each do |modulepath|
+        Dir[File.join(modulepath, '*', 'metadata.json')].sort.each do |metadata_json|
+          metadata = JSON.load(File.read(metadata_json))
+          next unless metadata['requirements'] && metadata['requirements'].is_a?(Array)
+
+          metadata['requirements'].select { |req| req['name'] == 'puppet' }.each do |req|
+            checks << versioncmp(metadata['name'], req['version_requirement'])
```

Comment:
Thanks, I've added a check for `version_requirement` into the `select` on the previous line, so this block will only run if both are set.